dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/ArtifactProblem.java

```diff
@@ -78,4 +81,44 @@ public static String formatProblems(Iterable<ArtifactProblem> problems) {
     }
     return output.toString();
   }
+
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (other == null || getClass() != other.getClass()) {
+      return false;
+    }
+
+    ArtifactProblem otherProblem = (ArtifactProblem) other;
+    return Objects.equals(artifact, otherProblem.artifact)
+        && sameDependencyPath(dependencyPath, otherProblem.dependencyPath);
+  }
+
+  private static boolean sameDependencyPath(
```

Comment:
Any chance this method belongs in DependencyPath?

Reply:
`com.google.cloud.tools.opensource.dependencies.DependencyPath` does not use `ImmutableList<DependencyNode>`. It uses `List<Dependency>`. So this method should be in DependencyPath.

Reply:
This method eeems like it works with both mutable and immutable lists.

Reply:
Yes. Updated. Also renamed to `equalsOnDependencies` as "same" implies checking object identity.

```
  private static boolean equalsOnDependencies(
      List<DependencyNode> listA, List<DependencyNode> listB) {
```